### PR TITLE
Add a bool check to stop the data being sent more than once

### PR DESCRIPTION
**Pull request by nitro-marky**
## What does this change?

Adds a boolean to check if the post has already been fired, hopefully this will stop the duplicates we seem to be getting.

### CODE REVIEW

File: src/web/browser/coreVitals/coreVitals.ts [hunk 1]
```diff
@@ -52,20 +63,16 @@ export const coreVitals = (): void => {
...
+		// Set page view and browser ID
+		if (window.guardian.config.ophan.browserId && window.guardian.ophan) {
+			jsonData.page_view_id = window.guardian.ophan.pageViewId;
+			jsonData.browser_id = window.guardian.config.ophan.browserId;
+		}
 
+		// Check if POST has been sent
+		if (!hasBeenSent) {
+			// If CLS has been calculated
+			if (jsonData.cls !== null) {
```
Comment (oliverlloyd) [inline src/web/browser/coreVitals/coreVitals.ts:69]: You can group these two conditional I presume?

File: src/web/browser/coreVitals/coreVitals.ts [hunk 2]
```diff
@@ -27,6 +27,16 @@ const jsonData: CoreWebVitalsPayload = {
...
+let hasBeenSent = false;
+
+// Can be tested locally or on prod
+const endpoint =
+	window.location.hostname === 'm.code.dev-theguardian.com' ||
+	window.location.hostname === 'localhost' ||
+	window.location.hostname === 'preview.gutools.co.uk'
+		? 'http://performance-events.code.dev-guardianapis.com/core-web-vitals'
+		: 'https://performance-events.guardianapis.com/core-web-vitals';
+
```
Comment (oliverlloyd) [inline src/web/browser/coreVitals/coreVitals.ts:39]: Did you mean to include these changes? They don't seem related to this PR?

File: src/web/browser/coreVitals/coreVitals.ts [hunk 3]
```diff
@@ -52,20 +63,19 @@ export const coreVitals = (): void => {
...
+			if (jsonData.cls !== null) {
+				// Set page view and browser ID
+				if (
+					window.guardian.config.ophan.browserId &&
+					window.guardian.ophan
+				) {
+					jsonData.page_view_id = window.guardian.ophan.pageViewId;
+					jsonData.browser_id =
+						window.guardian.config.ophan.browserId;
+				}
```
Comment (oliverlloyd) [inline src/web/browser/coreVitals/coreVitals.ts:78]: Unrelated changes?

File: src/web/browser/coreVitals/coreVitals.ts (hunk 1 above)
Comment (nitro-marky) [reply src/web/browser/coreVitals/coreVitals.ts:69]: Sure, I just did it as an initial readability thing

File: src/web/browser/coreVitals/coreVitals.ts (hunk 2 above)
Comment (nitro-marky) [reply src/web/browser/coreVitals/coreVitals.ts:39]: They were in before, I just moved them. It means it can be tested locally with no changes

**Comment by nitro-marky** [comment]
Sending one piece of data is a lot easier in the long term than trying to remove duplicates in the SQL side

**Comment by nitro-marky** [comment]
There are still moving pieces at the moment so it's not in a final form

**Comment by oliverlloyd** [comment]
> Sending one piece of data is a lot easier in the long term than trying to remove duplicates in the SQL side

My feeling on this is that actually removing queries in an SQL query should be trivial? I could be wrong on that though, you would need some form of page view id to group on - but assuming you _can_ identify the page view, then it should be simple to group on that.

My fear here is we're opening ourselves up the kind of bugs where if the core vitals lib reports data in a different way to what we expect in the wild, then we lose data. By keeping the clientside code simple we reduce that exposure

**Comment by nitro-marky** [comment]
I'll revert the changes and leave it to data
